### PR TITLE
started vbridge class and tests

### PR DESCRIPTION
**Pull request by saiakilesh**

### CODE REVIEW

File: lib/vbridge/VBridge.ts [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+      // Check the addresses of the asset. If it is zero, deploy a native token wrapper
+      let allowedNative: boolean = false;
+      for (const tokenToBeWrapped of vBridgeInput.vAnchorInputs.asset[chainID]!) {
+        // If passed '0' or zero address, token to be wrapped should support native.
+        if (checkNativeAddress(tokenToBeWrapped)) {
+          allowedNative = true;
+        }
+      }
+
+      let tokenInstance: GovernedTokenWrapper = await GovernedTokenWrapper.createGovernedTokenWrapper(
```
Comment (drewstone) [inline lib/vbridge/VBridge.ts:207]: We should support deploying a bridge with an existing token already too. For example, we already have a bridge for webbETH-test-1 live on ETH testnets and can use this same token for the variable bridge now, as long as we give the proper minting permissions to the VAnchor

File: lib/vbridge/VBridge.ts [hunk 2]
```diff
@@ -163,7 +163,7 @@ class VBridge {
   //   return new Bridge(bridgeConfig.bridgeSides, bridgeConfig.webbTokenAddresses, linkedAnchors, bridgeConfig.anchors);
   // }
 
-  public static async deployVBridge(vBridgeInput: VBridgeInput, deployers: DeployerConfig): Promise<VBridge> {
+  public static async deployVBridge(vBridgeInput: VBridgeInput, deployers: DeployerConfig, tokenInstance?: GovernedTokenWrapper): Promise<VBridge> {
```
Comment (saiakilesh) [inline lib/vbridge/VBridge.ts:166]: I added `tokenInstance` as an optional input to `deployBridge`, is this what you had in mind?

File: lib/vbridge/VAnchor.ts [hunk 3]
```diff
@@ -553,9 +555,12 @@ class VAnchor {
       input.extDataHash.toString()
     );
     //console.log(`current root (class) is ${toFixedHex(this.tree.root())}`);
-    outputs.forEach((x) => {
+    outputs.forEach(async (x) => {
```
Comment (drewstone) [inline lib/vbridge/VAnchor.ts:558]: What's async inside this loop? Also there's nuance using async with loops that don't work with for each afaik. If you need to you should use .map and wrap it in a await Promise.all(array.map(async (a) ...))

**Review by drewstone — CHANGES_REQUESTED** [review]

File: lib/vbridge/VAnchor.ts (hunk 3 above)
Comment (saiakilesh) [reply lib/vbridge/VAnchor.ts:558]: Yeah not sure, I'm not sure why I put async there. I will take it out.

**Review by drewstone — CHANGES_REQUESTED** [review]
Not ready, tremendous amount of commented out code. Please address. Would also recommend breaking transact apart so as to reduce code duplication.

**Pull request merged**